GVRf/Framework/framework/src/main/java/org/gearvrf/x3d/AnimationInteractivityManager.java

```diff
@@ -628,17 +663,122 @@ public void onSensorEvent(SensorEvent event) {
+                                                        .getSceneObjectByName(interactiveObject.getDefinedItem().getName());
+                                                GVRComponent gvrComponent = gvrSceneObject.getComponent(GVRLightBase.getComponentType());
+                                                gvrComponent.setEnable(eventUtility.getToggle());
+                                            }
+                                        }
+                                    }
+                                    else {
+                                        // this is an fbx or other non-X3D file for animation
+                                        if ( !activeFBXanimation ) {
+                                            InlineObject inlineObject = interactiveObjectFinal.getInlineObject();
+                                            GVRSceneObject inlineSceneObject = inlineObject.getInlineGVRSceneObject();
+                                            GVRAnimator animator = (GVRAnimator) inlineSceneObject.getComponent(GVRAnimator.getComponentType());
+                                            if ( animator != null ) {
+                                                List<GVRAnimation> gvrAnimations = animator.getAnimationList();
+                                                GVRAnimation gvrAnimation = null;
+
+                                                if  ( (inlineObject.getBeginKeyFrame() == InlineObject.UNSET_KEY_FRAME) || (inlineObject.getEndKeyFrame() == InlineObject.UNSET_KEY_FRAME)) {
+                                                    if ( (inlineObject.getBeginKeyFrame() == InlineObject.UNSET_KEY_FRAME) && (inlineObject.getEndKeyFrame() == InlineObject.UNSET_KEY_FRAME)) {
+                                                        // use the original FBX animation.
+                                                        gvrAnimation = gvrAnimations.get(0);
+                                                    }
+                                                    else {
+                                                        Log.e(TAG, "Error: ROUTE specifying the keyFrame range - only 1 may be an integers");
+                                                    }
+                                                }
+                                                else {
+                                                    // FBX animation specified in ROUTE as a set of keyFrames
+                                                    // play only a subset of the animation based on keyframes
+                                                    gvrAnimation = interactiveObjectFinal.getFBXAnimation();
+                                                    // check if we already copied over the animation
+                                                    if (gvrAnimation == null) {
+                                                        GVRAnimation gvrOriginalAnimation = gvrAnimations.get(0);
```

Comment:
My tests have animations that have translation and scale, but no rotation.  Or rotation with translation, and no scale, and they all passed.  Tests cases in the tests repo.

I agree this needs to be tested in case they use an FBX file with no animations, yet have an X3D file that tries to begin an animation with none there.